### ISSUE TITLE
Configure Push to Talk in Settings Panel

### ISSUE THREAD

**Issue by Xylemon** [qTox/qTox#903]
I think the user should be able to switch between normal voice settings and push to talk within qTox itself. Along with the option to change the keys for push to talk activation.

**Comment by lance0** [comment]
this is a skype alternative. use mumble if you want ptt.

**Comment by Xylemon** [comment]
I don't think that really helps anybody when you tell them to use another program. Especially when the point of an issue track is to suggest improvements and fixes for programs. Also qTox has push to talk already, so is asking for the ability to configure it in the GUI rather than through code really a such an odd request?

**Comment by lance0** [comment]
I don't think it's an improvement.

**Comment by Epictek** [comment]
Could you please explain how this is not an improvement? I have a mechanical keyboard so I prefer to use push to talk in group calls and having to use P as a push to talk key sounds horrible.

**Comment by zetok** [comment]
@lance0 No one cares that you don't think it's an improvement. If you don't like it, don't use it.

@Epictek plan is to have it configurable, and global. And recently it was changed to `CTRL + P`

**Comment by DanTheBritish** [comment]
Bump, introduced a few gaming friends to qTox and annoyed the heck out of them because no PTT.

**Comment by zetok** [comment]
@LapFox there is support for _PPT_, but not a global one.. (i.e. without qTox window being focused it won't work yet)

https://github.com/tux3/qTox/wiki/Keyboard-shortcuts

**Comment by Zer0-One** [comment]
+1 for a configurable PTT button. As-is, PTT is useless to me, because I can't remap the key.

**Comment by DanTheBritish** [comment]
Yeah, in that case a configurable one would be good, and a global one would be even better

**Comment by arrehn** [comment]
Push-to-talk is an important feature for privacy, security, performance, and clarity. It allows you to control only the sounds you want transmitted to be transmitted, keeping other ongoing unwanted side conversations or exclamations private to yourself, as well as avoiding transmission of incidental noises such as throat cleering, sneezes, eating and drinking, etc. It also has the effect of saving bandwith and keeping the conversation more clear and noise-free for other participants.

This is an important feature for a security and privacy-conscious platform like tox uses, and is certainly more convenient than clicking a small mute button. Please implement this feature.

**Comment by agilob** [comment]
#1880

**Comment by doncatnip** [comment]
I too would like such feature. I might have a look at this if I find the time.
@lance0 such "answers" are completely uncalled for.

**Comment by ghost** [comment]
+1, push to talk and the low latency is why I use Mumble, having push to talk in Tox would be great for the reasons mentioned above (clarity, bandwidth, privacy).

**Comment by anthonybilinski** [comment]
I've been looking into this. It looks like there's no built-in Qt solution for capturing key events when your window doesn't have focus. There's [libqxt](https://bitbucket.org/libqxt/libqxt/wiki/Home) which offers this support, but adding a large unmaintained lib just for this seems like a lot of maintenance work. I also found [UGlobalHotkey](https://github.com/falceeffect/UGlobalHotkey), a stand-alone lib which offers global hotkey capture. I've compiled a little test app with it locally and tested both X11 and Wayland Linux and it works great. Let me know if people have a problem with adding this dependency. Until that happens, I'll keep trying to work this into qTox, which would allow a global key push-to-talk (might make our PTT user configurable too) for both group calls and solo calls.
